modes/settings_menues.h

```diff
@@ -91,7 +91,7 @@ struct ChangeBladeLengthBlade1 : public SPEC::MenuBase {
     showlen_.Stop(blade());
   }
   void say() override {
-    getSL<SPEC>()->SayWhole(this->pos_);
+    getSL<SPEC>()->SayWhole(this->pos_ + 1);
```

Comment:
You'll also need to change where I set pos_ in mode_activate.

Reply:
oh, do you mean the line `if (len == -1) len = maxlen;` should be check if zero instead?
But everything in mode_activate() and size() is all still zero-based indexing at that point so it works correctly as-is.

So if we just set getLength() to this->pos_ + 1 , that propagates through for ShowLengthStyle and all is correct to the user.
Then Saying the number as we see it makes sense too with SayWhole(this->pos_ + 1).
Blade length gets saved to the SD as zero-based index as well (as normal?)
it's working, so it seems right?

Reply:
I was misreading your change I think.
I thought that the length was generally 1-indexed and tha pos_ is 0-indexed, but I guess that's not what is happening, instead we are changing the SayWhole() and what ShowLengthStyle is showing.

So, with that in mind, I think that this line is fine, but I have some comments for the other change...

Reply:
Actually I think you're right (although pos_ was not being set in here).
If  pos_ *is* set to `len` here, then I think all else is synced. see below.